server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -761,43 +687,43 @@ private static Entry createClone(
             Map<RepositoryShardId, ShardSnapshotStatus> shardStatusByRepoShardId
         ) {
             return new Entry(
+                state,
                 snapshot,
                 true,
                 false,
-                state,
+                Map.of(),
                 indices,
+                Map.of(),
                 List.of(),
                 List.of(),
                 startTime,
                 repositoryStateId,
-                Map.of(),
-                failure,
-                Map.of(),
                 version,
                 source,
                 shardStatusByRepoShardId,
                 Map.of(),
+                failure,
                 false
             );
         }
 
-        private Entry(
```

Comment:
IMO there was some value in hiding the constructor of this class. It has tricky invariants and we do not want callers to just construct arbitrary values. Could you add the constructor to forbidden APIs?